### PR TITLE
Added migration manual versioned ordering

### PR DESCRIPTION
**Pull request by jacksondelametter**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,8 +71,93 @@ one file, `original-specification-example.foo` alone.
...
+                 "name": "migration_3" 
+           },
+           {
+                 "name": "migration_4"
+           }
+        ]
+    }
+ ]
+```
+If Baton is executed, migrations would be ordered in which they are specified (migration_1, migration_2, migration_3, migration_4)
```
Comment (aaron-gary) [inline README.md]: Re-wording suggestion:

If Baton is executed on an ordered migration json, then migrations would be processed in the order in which they were specified (migration_1, migration_2, migration_3, migration_4)

File: README.md [hunk 2]
```diff
@@ -71,8 +71,93 @@ one file, `original-specification-example.foo` alone.
...
+           },
+           {
+                 "name": "migration_4", 
+                 "version": "5.0.0" 
+           }
+        ]
+    }
+ ]
+```
+If Baton is executed, migrations would be ordered by ascending version number (migration_3, migration_2, migration_1, migration_4)
```
Comment (aaron-gary) [inline README.md]: Re-wording suggestion:

If Baton is executed on a versioned migration json, then migrations would be processed by the version number in ascending order (migration_3, migration_2, migration_1, migration_4)

File: baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java [hunk 3]
```diff
@@ -156,21 +172,45 @@ protected void defaultFileSets() {
...
+        }
+    }
+    
     BatonExecutionSummary performMigration(Collection<GroupTarget> groupTargets) {
         BatonExecutionSummary executionSummary = new BatonExecutionSummary();
         for(GroupTarget groupTarget : groupTargets) {
             GroupSummary groupSummary = new GroupSummary(groupTarget.getGroup());
             for (MigrationTarget migrationTarget : groupTarget.getMigrations()) {
-                if (isActive(migrationTarget)) {
+                if (isActive(migrationTarget) && isMinimumVersion(migrationTarget, groupTarget.getType())) {
```
Comment (aaron-gary) [inline baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java:213]: Is it important to notify the user if a migration doesn't meet the minimum version criteria?  i.ie log a warning if it doesn't pass

**Review by aaron-gary — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jacksondelametter) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (jacksondelametter) [reply README.md]: Fixed.

**Pull request merged**